src/pages/SingleTopicPage.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getTopicsByTopicType } from "../utils/fetch";
+import SingleArticle from "../components/SingleArticle";
+
+const SingleTopicPage = () => {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true)
+  const { topictype } = useParams();
+
+  useEffect(() => {
+    getTopicsByTopicType(topictype).then((response) => {
+      setArticles(response.data.articles);
+      setIsLoading(false)
+    });
+  }, []);
+
+  if (isLoading) return <p>Loading...</p>;
+
+  return (
+    <div className="articleListDiv">
+      {articles.map((article) => {
+        return (
+          <SingleArticle
+            key={article.article_id}
+            article={article}
+          />
+        );
+      })}
+    </div>
+  );
+};
```

Comment:
This works, but it basically does the same job as ArticlesList. You also have two axios requests that go to /articles, just one has a query attached.

You can reuse the ArticlesList component for the topics/:topics_type route and use the same getArticles axios function you used before. Axios requests take a second argument of params, have a look at the notes, if the parameter is there it will use it, otherwise it will ignore it and retrieve all articles. 

If you're able to reuse that same ArticlesList component for both routes, you will save yourself a lot of code that is repeating itself.